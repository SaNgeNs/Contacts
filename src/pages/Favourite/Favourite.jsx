import React, { memo }  from 'react';
import Helmet from 'react-helmet/lib/Helmet';
import CardList from 'Components/CardList';
import { useSelector } from 'react-redux';

export const Favourite = () => {
  const cards = useSelector(state => {
    const favourite = [];

    state.cards.ids.forEach(id => {
      const allCards = state.cards.entities;

      if (allCards[id].isFavourite) {
        favourite.push(allCards[id]);
      }
    });

    return favourite;
  });

  return (
    <div>
      <Helmet>
        <title>Favourite</title>
        <meta property="og:title" content="Favourite"/>
        <meta name="twitter:title" content="Favourite"/>
        <meta name="og:image:alt" content="Favourite"/>
        <link rel="canonical" href="https://test.com" />
        <meta property="og:url" content="https://test.com" />
        <meta httpEquiv="x-dns-prefetch-control" content="on" />
        <meta name="theme-color" content="#ffffff" />
        <meta property="og:site_name" content="Favourite" />
        <meta property="og:type" content="page" />
        <meta name="description" content="Favourite" />
        <meta property="og:description" content="Favourite" />
      </Helmet>

      <h1
        style={{
          padding: '0 10px'
        }}
      >
        Favourite:
      </h1>

      <CardList
        cards={cards}
      />
    </div>
  );
};

export default memo(Favourite);
