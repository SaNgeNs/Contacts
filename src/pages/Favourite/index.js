import withErrorHandler from 'Components/withErrorHandler';
import Favourite from './Favourite';

export default withErrorHandler(Favourite);
