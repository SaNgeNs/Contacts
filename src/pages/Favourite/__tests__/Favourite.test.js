import React from 'react';
import customRender from 'Mock/customRender';
import MockStore from 'Mock/store';

import Favourite from '../Favourite';

describe('<Favourite />', () => {
  it('should render favourite page', () => {
    const { container, unmount } = customRender(
      <Favourite />,
      {},
      MockStore,
    );

    expect(container).toMatchSnapshot();
    unmount();
  });
});
