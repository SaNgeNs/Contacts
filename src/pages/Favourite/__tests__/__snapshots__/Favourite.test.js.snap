// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favourite /> should render favourite page 1`] = `
<div>
  <div>
    <h1
      style="padding: 0px 10px;"
    >
      Favourite:
    </h1>
  </div>
</div>
`;
